tools/infrasys/src/keys.rs

```diff
@@ -0,0 +1,130 @@
+use async_trait::async_trait;
+use pubsys_config::{KMSKeyConfig, SigningKeyConfig};
+use rusoto_cloudformation::{CloudFormation, CloudFormationClient, CreateStackInput};
+use rusoto_core::Region;
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+use std::str::FromStr;
+
+use super::{error, shared, Result};
+
+/// Creates keys using data stored in SigningKeyConfig enum
+/// Output: Edits KMSConfig fields in place after creating new keys
+pub async fn create_keys(signing_key_config: &mut SigningKeyConfig) -> Result<()> {
+    // An extra check even through these parameters are checked earlier in main.rs
+    check_signing_key_config(signing_key_config)?;
+    match signing_key_config {
+        SigningKeyConfig::file { .. } => (),
+        SigningKeyConfig::kms { config, .. } => {
+            config
+                .as_mut()
+                .context(error::MissingConfig {
+                    missing: "config field for a kms key",
+                })?
+                .create_kms_keys()
+                .await?;
+        }
+        SigningKeyConfig::ssm { .. } => (),
+    }
+    Ok(())
+}
+
+pub fn check_signing_key_config(signing_key_config: &SigningKeyConfig) -> Result<()> {
+    match signing_key_config {
+        SigningKeyConfig::file { .. } => (),
+        SigningKeyConfig::kms { config, .. } => {
+            let unwrapped_config = config.as_ref().context(error::MissingConfig {
```

Comment:
We can just rebind the variable to be `config`.
```suggestion
            let config = config.as_ref().context(error::MissingConfig {
```